src/formatDate.js

```diff
@@ -50,7 +50,57 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const CENTURY_21 = '20';
+  const CENTURY_20 = '19';
+  const dateArr = date.split(`${fromFormat[3]}`);
+  const expectedDate = [];
+  let month, day, shortYear, longYear;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i]) {
+      case 'MM':
+        month = dateArr[i];
+        break;
+      case 'DD':
+        day = dateArr[i];
+        break;
+      case 'YY':
+        shortYear = dateArr[i];
+        break;
+      case 'YYYY':
+        longYear = dateArr[i];
+        break;
+    }
+  }
+
+  if (shortYear === undefined) {
+    shortYear = longYear.slice(2, 4);
+  } else if (longYear === undefined) {
+    if (shortYear < 30) {
+      longYear = CENTURY_21 + shortYear;
+    } else {
+      longYear = CENTURY_20 + shortYear;
+    }
+  }
```

Comment:
If you need to check that value doesn't exist you can use just Logical NOT (!).
!shortYear instead of shortYear === undefined.
Also, it's a bad practice to use "else if" and nested "if" conditions.
Try to get rid of them.